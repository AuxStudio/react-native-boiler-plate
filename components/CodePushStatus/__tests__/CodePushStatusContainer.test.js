import React from 'react';
import renderer from 'react-test-renderer';
import codePush from 'react-native-code-push';

import { CodePushStatusContainer } from '..';

describe('CodePushStatusContainer', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<CodePushStatusContainer />);

      expect(component).toMatchSnapshot();
    });

    it('renders the up to date state', () => {
      const component = renderer.create(
        <CodePushStatusContainer codePushSyncStatus={codePush.SyncStatus.UP_TO_DATE} />,
      );

      expect(component).toMatchSnapshot();
    });

    it('renders the update ready state', () => {
      const component = renderer.create(
        <CodePushStatusContainer codePushSyncStatus={codePush.SyncStatus.UPDATE_INSTALLED} />,
      );

      expect(component).toMatchSnapshot();
    });

    it('renders the checking for updates state', () => {
      const component = renderer.create(
        <CodePushStatusContainer codePushSyncStatus={codePush.SyncStatus.CHECKING_FOR_UPDATE} />,
      );

      expect(component).toMatchSnapshot();
    });

    it('renders the downloading update state', () => {
      const codePushDownloadProgress = { receivedBytes: 100, totalBytes: 200 };
      const component = renderer.create(
        <CodePushStatusContainer
          codePushSyncStatus={codePush.SyncStatus.DOWNLOADING_PACKAGE}
          codePushDownloadProgress={codePushDownloadProgress}
        />,
      );

      expect(component).toMatchSnapshot();
    });
  });
});
