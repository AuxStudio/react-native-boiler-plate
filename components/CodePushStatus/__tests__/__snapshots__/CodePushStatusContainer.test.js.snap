// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodePushStatusContainer renders renders the checking for updates state 1`] = `null`;

exports[`CodePushStatusContainer renders renders the downloading update state 1`] = `null`;

exports[`CodePushStatusContainer renders renders the up to date state 1`] = `null`;

exports[`CodePushStatusContainer renders renders the update ready state 1`] = `null`;

exports[`CodePushStatusContainer renders renders with minimum required props 1`] = `null`;
